packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -164,9 +177,40 @@ class OverlayEntry extends ChangeNotifier {
   ///
   /// You need to call this function if the output of [builder] has changed.
   void markNeedsBuild() {
+    assert(!_disposedByOwner);
     _key.currentState?._markNeedsBuild();
   }
 
+  void _didUnmount() {
+    if (_disposedByOwner && !mounted) {
```

Comment:
from the method name (_didUnmount) it should follow that mounted must always be false here? Should that `!mounted` just be in an assert?